pages/api/paybutton/index.ts

```diff
@@ -1,48 +1,38 @@
 import { NextApiRequest, NextApiResponse } from 'next'
-import { PayButton } from 'types'
 import * as paybuttonsService from 'services/paybuttonsService'
 import { validateAddresses } from 'utils/validators'
+import { RESPONSE_MESSAGES } from 'constants/index'
 
-const generateIdFromUserId = userId => Math.random().toString(16).slice(2)
 
-
-const fetchResource = (userIdFromQuery: string): PayButton[] => {
-  const userId = userIdFromQuery || "test-user-id"
-  const temp_addresses = ['ecash:qpz274aaj98xxnnkus8hzv367za28j900c7tv5v8pc', 'bitcoincash:qrw5fzqlxzf639m8s7fq7wn33as7nfw9wg9zphxlxe']
-  return  [{
-    id: generateIdFromUserId(userId),
-    userId,
-    addresses: temp_addresses
-  }]
-}
-
-
-export default (req: NextApiRequest, res: NextApiResponse) => {
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  const values = req.body
+  const userId = values.userId
+  if (!userId) {
+    res.status(400).json(RESPONSE_MESSAGES.USER_ID_NOT_PROVIDED_400)
+    return
```

Comment:
Do we need this return statement here?

Reply:
Yes, or the paybutton will be created without a providerUserId.

Reply:
can we move all errors to the exception handling block? 

in this case would be something like `throw new Error(RESPONSE_MESSAGES.USER_ID_NOT_PROVIDED)`

and then on catch block:
`switch (e.message): 
    case RESPONSE_MESSAGES.USER_ID_NOT_PROVIDED:
           res.status(400).json......
           
 
 we could throw an error on validateAddresses as well and add the response on a switch case

Reply:
I think for now we can just return res.status... instead of just a void return call

Reply:
@alcipir I've change the logic on the endpoint to throw errors on invalid requests, and a `switch` case to  adapt the response accordingly.